### PR TITLE
exclude lr scheduler's state from accumulators_holder

### PR DESCRIPTION
**Pull request by iclementine**
### PR types

Bug fixes

### PR changes

APIs

### Describe

Fix the bug that "an optimizer that has a learning rate scheduler who has called `set_state_dict()` before any update would cause an error in `step()` afterwards."

### CODE REVIEW

File: python/paddle/optimizer/optimizer.py [hunk 1]
```diff
@@ -286,6 +286,10 @@ def set_state_dict(self, state_dict):
         if isinstance(self._learning_rate, LRScheduler):
             self._learning_rate.set_state_dict(state_dict["LR_Scheduler"])
 
+        # NOTE: exclude learning rate scheduler's state from 
+        # _accumulators_holder.
+        if "LR_Scheduler" in state_dict:
+            state_dict.pop("LR_Scheduler")
```
Comment (guoshengCS) [inline python/paddle/optimizer/optimizer.py:293]: 这里是否最好是重新创建一个state_dict以避免对输入的修改呢

Comment (iclementine) [reply python/paddle/optimizer/optimizer.py:293]: 好的

Comment (iclementine) [reply python/paddle/optimizer/optimizer.py:293]: 已经修改

**Review by guoshengCS — APPROVED** [review]
LGTM

**Review by chenwhql — APPROVED** [review]
LGTM
